### DISCUSSION TITLE
Improve analyzation of full storage issues

### DISCUSSION THREAD

**Discussion by Sebbl22** [datalust/seq-tickets#1876] [Ideas]
Hi,
we have a single company-wide Seq server which is used by multiple teams. We have 100+ API key, which ingest to Seq.
Sometimes a single application is out of control and produces a very large amount of messages. If this happens during business hours, we can use Data --> Ingestion view and sort by "Current Ingestion". If it didn't happen during business hours and the issue is resolved, we have no easy way to analyze which API key ingested how much data, because Data --> ingestion only displays a limited history.
Usually we would use a query like `select count(*) as c from stream group by Environment, Application order by c desc` which can run very long.

I would like to see some features implemented regarding this issue like some of those:
- Allow sorting the Data --> Ingestion view by messages/min AND by data/min. Currenty, messages/min and data/min is displayed in one column and it can only be filtered by messages/min
- Implement a history to Data --> Ingestion, so that past ingestion could be analyzed easily.
- Alternatively, implement a single number each for No. of messages and amount of data per API key, so that a Seq admin can see, how much disk space is occupied by which API key.

I think I wrote a similar issue some months ago, but I could not find it. Sorry for opening another issue.

**Comment by KodrAus** [comment]
Thanks for the suggestions @Sebbl22 :+1:

A longer history on the ingestion page was something we originally discussed when implementing it, but left for a future milestone.

As an aside, if you are finding yourself with query performance issues running queries please feel free to reach out to use at `support@datalust.co` and we can take a look.
